pkg/manifests/config.go

```diff
@@ -16,19 +16,32 @@ package manifests
 
 import (
 	"bytes"
+	"context"
 	"encoding/json"
 	"fmt"
 	"io"
+	"math"
 
 	configv1 "github.com/openshift/api/config/v1"
 	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	poperator "github.com/prometheus-operator/prometheus-operator/pkg/operator"
 	v1 "k8s.io/api/core/v1"
 	k8syaml "k8s.io/apimachinery/pkg/util/yaml"
 	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
+	"k8s.io/klog/v2"
 )
 
 const (
 	DefaultRetentionValue = "15d"
+
+	// Limit the body size from scrape queries
+	// Assumptions: one node has in average 110 pods, each pod exposes 400 metrics, each metric is expressed by on average 250 bytes.
+	// 1.5x the size for a safe margin, it rounds to 16MB (16,500,000 Bytes).
+	minimalSizeLimit = 1.5 * 110 * 400 * 250
```

Comment:
We probably need a "safe" lower-bound value for the "automatically computed"  value but I'm not sure that this value is accurate. I would take a typical CI cluster, load it with as many pods/secret/configmaps as possible and measure the body size returned by kube-state-metrics `/metrics`.

Reply:
Having tested with 100k secrets and 100k config maps with kube-burner, KSM scrape target gives us ~32MB body size. 
So this lower bound is tripled to 48MB in new commit. It should be large enough I suppose :) 

![Screenshot from 2022-05-10 12-09-02](https://user-images.githubusercontent.com/6040960/167638814-b89cfc9f-bd56-489a-abc5-11c415a56420.png)